### DISCUSSION TITLE
.Net 5.0 using older SDK, Core, Standard packages

### DISCUSSION THREAD

**Discussion by xbotuk** [dotnet/designs#240] [Q&A] [answered]
I would like to know if .Net 5.0 can use libraries from older SDK, Core, Standard packages easily. If so how and which older versions will be supported?

**Comment by jnm2** [comment, marked as answer]
A project targeting .NET 5 can use libraries which target:

- .NET 5
- Any version of .NET Core (1.0–3.1)
- Any version of .NET Standard (1.0–2.1)
- Any version of .NET Framework (1.0–4.8) BUT with a build warning and no guarantee against MissingMethodException if the library tries to access something that is in .NET Framework that is not in .NET 5. https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer can tell you whether the library might do this.

> If so how

The same way you would reference a .NET 5 library. Using `<PackageReference Include="SomePackageId" Version="1.2.3" />` is the standard way, but you can also use `<Reference Include="Path/to/SomeAssemblyFile.dll" />`.
